### PR TITLE
remove python2 and rename swig bindings

### LINKED ISSUE #3186: rename binding swig_python to python
Currently, `cmake` (without setting PLUGINS or BINDINGS) outputs:

```
-- Exclude Plugin python because swig_python binding is required
-- Exclude Plugin python2 because swig_python2 binding is required
[...]
-- Include Binding cpp
-- Exclude Binding jna because javac (java compiler) not found, which is only included in a jdk
-- Exclude Binding rust because excluded by category EXPERIMENTAL
-- Exclude Binding swig_ruby because explicitly excluded
-- Exclude Binding gsettings because excluded by category EXPERIMENTAL
-- Exclude Binding intercept_fs because excluded by category EXPERIMENTAL
-- Include Binding intercept_env
-- Exclude Binding io_uv because excluded by category EXPERIMENTAL
-- Exclude Binding io_ev because excluded by category EXPERIMENTAL
-- Exclude Binding io_glib because excluded by category EXPERIMENTAL
```

It does not write anything about the python binding. (info about swig_ruby is also wrong).

To include the python binding, currently it is needed to call (see also #3185 for the confusion):

```
cmake -DSWIG_EXECUTABLE=/usr/bin/swig3.0 -DBINDINGS="cpp;swig_python" ../workspace
```

I propose following changes:

- [x] rename the swig_python and swig_lua bindings to simply python and lua
- [x] remove the python2 binding and plugin (as python2 days are counted anyway)
- [x] fix the output, that "Exclude Binding" is also printed correctly for python/lua/ruby
- [x] that the python binding is included by default, as it is well tested and in productive use

@manuelm what do you think?

### PR DESCRIPTION
**Pull request by manuelm**
Fixes #3186

lets see if this still builds...

### CODE REVIEW

File: doc/images/plugins.dot [hunk 1]
```diff
@@ -150,7 +150,6 @@ strict digraph G {
 	process [shape=Mdiamond];
 	profile [shape=Mdiamond];
 	python [shape=Mdiamond];
-	python2 [shape=Mdiamond];
```
Comment (mpranj) [inline doc/images/plugins.dot]: This is removed when running `scripts/dev/draw-all-plugins`, which also fixes the png automatically.

Comment (manuelm) [reply doc/images/plugins.dot]: there is a png? 🤯

**Comment by manuelm** [comment]
> Thank you for the PR! Is it ready?

No, not ready yet. Forgot about the label.

> Looks like the build almost went through. We'll need to remove `debian/python-elektra.install` and `debian/libelektra4-python2.install` from the debian branch.

Can you please do that?

**Comment by mpranj** [comment]
> Can you please do that?

Sure, I'll do that when the PR is ready.

**Comment by markus2330** [comment]
Thank you for the PR! :sparkle:

**Comment by markus2330** [comment]
@mpranj there are some files to be removed in the debian branch, can you do it? (directly push there)

**Comment by mpranj** [comment]
@markus2330 yes, I thought I would remove it when this PR is ready rather than immediately.

**Comment by markus2330** [comment]
jenkins build libelektra please

**Comment by markus2330** [comment]
> @markus2330 yes, I thought I would remove it when this PR is ready rather than immediately.

For removing earlier is easier, then the builds of the PR go through. For adding or renaming is better afterwards, as it would break master.

**Comment by manuelm** [comment]
from my short tests:

```
cmake /...
[...]
-- Include Binding python
-- Include Binding lua
-- Exclude Binding ruby because explicitly excluded
[...]
```

```
cmake -DSWIG_EXECUTABLE= /...
[...]
-- Search for swig2 instead
-- Exclude Binding python because neither swig2/3 found. Please install swig3.0 and set -DSWIG_EXECUTABLE=
-- Exclude Binding lua because neither swig2/3 found. Please install swig3.0 and set -DSWIG_EXECUTABLE=
-- Exclude Binding ruby because explicitly excluded
[...]
```

**Comment by markus2330** [comment]
jenkins build libelektra please

**Comment by mpranj** [comment]
@manuelm can you please rebase to fix the conflict? Its not urgent. I don't know if 5122b2848d1f3008e203b9642c5c104eabf1dc8d fixes the problems on Jenkins.

**Comment by mpranj** [comment]
Seems like I broke builds with 5122b2848d1f3008e203b9642c5c104eabf1dc8d. I resolved the merge conflict such that we can merge this PR sooner.

**Comment by manuelm** [comment]
Thank you

**Comment by mpranj** [comment]
Thank you for maintaining the bindings and for the cleanup!

**Pull request merged**